### PR TITLE
Fixes #28210 - API to fix Locked template input

### PR DESCRIPTION
**Pull request by anandkumaragrawal**

### CODE REVIEW

File: app/models/template_input.rb [hunk 1]
```diff
@@ -62,6 +63,19 @@ def basic?
 
   private
 
+  def prevent_delete_if_template_is_locked
+    if template.locked
```
Comment (ofedoren) [inline app/models/template_input.rb]: On line 74 you check for template to not be `nil`, but here you don't. Isn't it better to check everywhere? Although I'd assume that template is present always. Template input shouldn't exist without a template, not sure if this constrait exists though...

File: app/models/template_input.rb [hunk 2]
```diff
@@ -62,6 +63,19 @@ def basic?
...
+      errors.add(:base, _("Cannot delete template input as template is locked ."))
```
Comment (ofedoren) [inline app/models/template_input.rb]: Nit: Extra space before the dot.

File: test/models/template_input_test.rb [hunk 3]
```diff
@@ -37,4 +37,33 @@ class TemplateInputTest < ActiveSupport::TestCase
     before { template_input.input_type = 'puppet_parameter' }
     it { assert template_input.puppet_parameter_template_input? }
   end
+
```
Comment (ofedoren) [inline test/models/template_input_test.rb:40]: Could you please add a test for deletion also?

**Review by ofedoren — COMMENTED** [review]
Thanks, @anandkumaragrawal! Overall looks good, I've tested CUD operations via hammer and they seem to be working as expected. Some nitpicks are inline though.

Adding @xprazak2 to confirm this fixes the issue (I don't have permissions to merge).

UPD: please, check the commit name, it should follow some rules: https://theforeman.org/handbook.html#Codingstandards

File: app/models/template_input.rb (hunk 1 above)
Comment (anandkumaragrawal) [reply app/models/template_input.rb]: Thanks, @ofedoren for reviewing this. 

I have fixed this as per the suggestion.

File: app/models/template_input.rb (hunk 2 above)
Comment (anandkumaragrawal) [reply app/models/template_input.rb]: Done as per the suggestion.

File: test/models/template_input_test.rb (hunk 3 above)
Comment (anandkumaragrawal) [reply test/models/template_input_test.rb:40]: Added test cases for deletion.

**Review by xprazak2 — APPROVED** [review]

**Comment by xprazak2** [comment]
Thanks everyone!

**Pull request merged**